modules/elasticsearch/elasticsearch_test.go

```diff
@@ -92,65 +92,40 @@ func TestElasticsearch(t *testing.T) {
 
 			// set the password for the request using the Authentication header
 			if tt.passwordCustomiser != nil {
-				if esContainer.Settings.Username != "elastic" {
-					t.Fatal("expected username to be elastic but got", esContainer.Settings.Username)
-				}
+				require.Equalf(t, "elastic", esContainer.Settings.Username, "expected username to be elastic but got: %s", esContainer.Settings.Username)
 
 				// basicAuthHeader {
 				req.SetBasicAuth(esContainer.Settings.Username, esContainer.Settings.Password)
 				// }
 			}
 
 			resp, err := httpClient.Do(req)
-			if resp != nil {
-				defer resp.Body.Close()
-			}
+			require.NotNil(t, resp)
+			defer resp.Body.Close()
+			require.NoError(t, err)
```

Comment:
bug: check for error first then no need to check for nil resp.
```suggestion
			require.NoError(t, err)
			defer resp.Body.Close()
```